staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -99,11 +114,21 @@ func (m *Helper) Delete(namespace, name string) (runtime.Object, error) {
 }
 
 func (m *Helper) DeleteWithOptions(namespace, name string, options *metav1.DeleteOptions) (runtime.Object, error) {
+	if options == nil {
+		options = &metav1.DeleteOptions{}
+	}
+	if m.DryRunServer {
+		if err := m.DryRunVerifier.HasSupport(m.GVK); err != nil {
+			return nil, err
+		}
+		options.DryRun = []string{metav1.DryRunAll}
+	}
+
 	return m.RESTClient.Delete().
 		NamespaceIfScoped(namespace, m.NamespaceScoped).
 		Resource(m.Resource).
 		Name(name).
-		Body(options).
+		VersionedParams(options, metav1.ParameterCodec).
```

Comment:
I would keep these options in the body, have you seen any regression?

Reply:
Oh the test that I saw fail is actually not in k8s.io/kubectl/pkg/cmd/delete, but here in k8s.io/cli-runtime/pkg/resource:

```
INFO: From Testing //staging/src/k8s.io/cli-runtime/pkg/resource:go_default_test:
==================== Test output for //staging/src/k8s.io/cli-runtime/pkg/resource:go_default_test:
--- FAIL: TestHelperDelete (0.00s)
    --- FAIL: TestHelperDelete/test3pkg/kubectl/genericclioptions/resource/helper_test.go (0.00s)
        helper_test.go:133: unexpected error: false no kind "DeleteOptions" is registered for version "" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:69"
```

It has to do with the scheme, but I'm not sure how this works yet.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cli-runtime/pkg/resource/helper_test.go#L34-L41

Reply:
Ok I actually specified WithoutConversion for the NegotiatedSerializer and that fixed the test. I don't know how this works!